packages/pds/src/api/com/atproto/sync/getRecord.ts

```diff
@@ -19,17 +20,27 @@ export default function (server: Server, ctx: AppContext) {
         }
       }
       // must open up the db outside of store interface so that we can close the file handle after finished streaming
-      const actorDb = await ctx.actorStore.db(did)
-      const storage = new SqlRepoReader(actorDb)
-      const commit = params.commit
-        ? CID.parse(params.commit)
-        : await storage.getRoot()
+      const actorDb = await ctx.actorStore.openDb(did)
 
-      if (!commit) {
-        throw new InvalidRequestError(`Could not find repo for DID: ${did}`)
+      let carStream: stream.Readable
+      try {
+        const storage = new SqlRepoReader(actorDb)
+        const commit = params.commit
+          ? CID.parse(params.commit)
+          : await storage.getRoot()
+
+        if (!commit) {
+          throw new InvalidRequestError(`Could not find repo for DID: ${did}`)
+        }
+        const carIter = repo.getRecords(storage, commit, [{ collection, rkey }])
+        carStream = byteIterableToStream(carIter)
+      } catch (err) {
+        await actorDb.close()
+        throw err
       }
-      const carIter = repo.getRecords(storage, commit, [{ collection, rkey }])
-      const carStream = byteIterableToStream(carIter)
+      carStream.on('error', actorDb.close)
+      carStream.on('close', actorDb.close)
```

Comment:
I think that `close()` may not be bound to `actorDb`, we might need something like this (but shouldn't be too bad to test):
```suggestion
      const closeDb = () => actorDb.close()
      carStream.on('error', closeDb)
      carStream.on('close', closeDb)
```

Reply:
yup yup good catch 👍